lib/compiler/passes/generate-js.js

```diff
@@ -705,7 +705,7 @@ function generateJS(ast, options) {
             break;
 
           default:
-            throw new Error("Invalid opcode: " + bc[ip] + ".");
+            throw new Error("Invalid opcode #" + ip + ": " + bc[ip] + ".");
```

Comment:
Like above, I’d add both the position and the rule into the message: “Invalid opcode: 77 (rule: Expression, position: 6).”.